.github/workflows/main.yml

```diff
@@ -43,10 +43,8 @@ jobs:
           python-version: ${{ matrix.python-version }}
           architecture: x64
 
-      - name: Setup Package and Install Devel Dependancies
-        run: |
-          python -m pip install .
-          python -m pip install -Ur dev-requirements.txt
+      - name: Install nox
+        run: python -m pip install nox
 
-      - name: Unit Tests
-        run: py.test tests -v
+      - name: Run nox
+        run: nox
```

Comment:
I would not make it as part of `CI`. why?
`nox` is just like `tox` which help to test package on a different `version of python`. Our CI already has taken care of supporting python versions and different OS.

Reply:
I added this to simplify setup and dependencies.
Currently only tests environment is added. If you want to install it we can write more code which can run with only one command like:
```
nox --session tests
nox --session development
nox --session user
```
[source](https://nox.thea.codes/en/stable/tutorial.html)

Reply:
@amolkahat I like your proposal, do some changes.

1. Restore workflow as it is.
2. add `nox` dependency in `dev-requirements.txt`
3. Let utilize nox features to test PR changes locally (not in CI); I mean parametrize over supported python versions like
```
@nox.session(python=[ "3.6", "3.7"])
def test(session):
    pass
```

Reply:
@digitronik why not in CI?